mesonbuild/interpreter/dependencyfallbacks.py

```diff
@@ -18,6 +20,7 @@
 class DependencyFallbacksHolder(InterpreterObject):
```

Comment:
This is not an `ObjectHolder` since it doesn't actually hold anything. Especially when looking at #8885 (which I hope will be merged very soon), it would be nice if this was renamed to something like `DependencyFallbacksObject`.

Even better would be if you could refactor this class to into a raw `DependencyFallbacksObject` with just the logic and *all* the data, and a `DependencyFallbacksHolder` with all the user-facing functions and only the `self.held_object`.

Reply:
Also, `DependencyFallbacksHolder` must derive from `MutableInterpreterObject` since it is mutable

Reply:
> Even better would be if you could refactor this class to into a raw `DependencyFallbacksObject` with just the logic and _all_ the data, and a `DependencyFallbacksHolder` with all the user-facing functions and only the `self.held_object`.

That makes a lot of sense, good idea. Made that change.

> Also, `DependencyFallbacksHolder` must derive from `MutableInterpreterObject` since it is mutable

It's not that easy because interpreterbase does deepcopy MutableInterpreterObject, which causes copy of self.interpreter.